lib/service/src/main/java/io/atlasmap/service/AtlasService.java

```diff
@@ -351,19 +373,24 @@ public Response listMappingBuilderClasses(@Context UriInfo uriInfo) {
     }
 
     /**
-     * Delete all user-defined library JAR files and mapping projects including Mapping Definitions and Documents saved on the server.
+     * Delete all user-defined library JAR files and mapping projects including Mapping Definitions and
+     * Documents associated with the specified mapping definition ID.
+     *
+     * @param mappingDefinitionId mapping definition ID
      * @return empty response
      */
     @DELETE
-    @Path("/all")
-    @Operation(summary = "Delete all", description = "Delete all user-defined library JAR files and mapping projects")
+    @Path("/all/{mappingDefinitionId}")
+    @Operation(summary = "Delete all", description = "Delete all user-defined library JAR files, documents and mapping projects")
     @ApiResponses({
-        @ApiResponse(responseCode = "200", description = "All user-defined libarary JARs and mapping projects were deleted successfully"),
-        @ApiResponse(responseCode = "204", description = "Unable to delete all user-defined JAR files and mapping projects") })
-    public Response deleteAll() {
-        LOG.debug("deleteAll");
+        @ApiResponse(responseCode = "200", description = "All user-defined libarary JARs,documents and mapping projects were deleted successfully"),
+        @ApiResponse(responseCode = "204", description = "Unable to delete all user-defined JAR files, documents and mapping projects") })
+    public Response deleteAll(
+        @Parameter(description = "Mapping Definition ID") @PathParam("mappingDefinitionId") Integer mappingDefinitionId) {
+        LOG.debug("deleteAll {} ", mappingDefinitionId);
         getLibraryLoader().clearLibraries();
         deleteAllMappingProjects();
```

Comment:
This `deleteAllMappingProjects()` is supposed to remove all documents, if not we should fix `deleteAllMappingProjects()` instead of introducing new REST endpoint.

Reply:
So the `AtlasService` REST endpoint `deleteAll()` is what the UI reset currently calls.  It calls `deleteAllMappingProjects()`.  So - i guess I should modify `deleteAllMappingProjects` to take a `mappingDefinitionId` param and to call the admHandler's `admHandler.deleteAllDocuments().

Reply:
Acutally because it is deleting libraries, it should delete all projects, IOW for all existing `mappingDefinitionId`, so it wouldn't take `mappingDefinitionId` as a parameter.
Looking at the `deleteAllMappingProjects()`, it aggressively deletes everything in the persistent directory. Does it happen to be just a cache problem, possibly the `deleteAllMappingProjects()` has to clear `admHandlerMap`?

Reply:
Hey @igarashitm - you're right as always - I re-instantiated the adm handler hash map and it fixed reset.  ref:
<pre>
diff --git a/lib/service/src/main/java/io/atlasmap/service/AtlasService.java b/lib/service/src/main/java/io/atlasmap/service/AtlasService.java
index bb8565e80..9520442c7 100644
--- a/lib/service/src/main/java/io/atlasmap/service/AtlasService.java
+++ b/lib/service/src/main/java/io/atlasmap/service/AtlasService.java
@@ -231,6 +231,7 @@ public class AtlasService extends BaseAtlasService {
         }
 
         AtlasUtil.deleteDirectoryContents(mappingFolderPathFile);
+        this.admHandlerMap = new ConcurrentHashMap<>();
         return Response.ok().build();
     }
</pre>

Reply:
Cool - I could guess because that's my bug :rofl: 
maybe `this.admHandlerMap.clear()` looks better